Field study browser

nahgnaw opened the pull request:
The new field study browser with a map view and the list result set hidden.

nahgnaw committed:
Create field study browser branch, and update ingest scripts.

nahgnaw committed:
Update ingest and mapping with field sites.

nahgnaw committed:
Add field Sites facet.

nahgnaw committed:
Show field studies as markers with popups on the map.

nahgnaw committed:
Show more field study info in the popups.

nahgnaw committed:
Add marker clusters.

olyerickson commented:
Link to a test version? Thanks!

On Friday, October 16, 2015, Han Wang notifications@github.com wrote:

> The new field study browser with a map view and the list result set
> 
> ## hidden.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/tetherless-world/dco-elasticsearch/pull/25
> Commit Summary
> - Create field study browser branch, and update ingest scripts.
> - Update ingest and mapping with field sites.
> - Add field Sites facet.
> - Show field studies as markers with popups on the map.
> - Show more field study info in the popups.
> - Add marker clusters.
> 
> File Changes
> - _A_ html/field-studies.html
>   https://github.com/tetherless-world/dco-elasticsearch/pull/25/files#diff-0
>   (308)
> - _A_ ingest/ingest-field-studies.py
>   https://github.com/tetherless-world/dco-elasticsearch/pull/25/files#diff-1
>   (459)
> - _A_ ingest/mappings/field-study.json
>   https://github.com/tetherless-world/dco-elasticsearch/pull/25/files#diff-2
>   (213)
> - _M_ ingest/queries/describeProject.rq
>   https://github.com/tetherless-world/dco-elasticsearch/pull/25/files#diff-3
>   (4)
> - _A_ ingest/queries/listFieldStudies.rq
>   https://github.com/tetherless-world/dco-elasticsearch/pull/25/files#diff-4
>   (17)
> 
> Patch Links:
> - https://github.com/tetherless-world/dco-elasticsearch/pull/25.patch
> - https://github.com/tetherless-world/dco-elasticsearch/pull/25.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/tetherless-world/dco-elasticsearch/pull/25.

## 

John S. Erickson, Ph.D.
Director of Operations, The Rensselaer IDEA
Deputy Director, Web Science Research Center (RPI)
http://tw.rpi.edu olyerickson@gmail.com
Twitter & Skype: olyerickson

nahgnaw committed:
Remove marker clustering.

zednis commented on ingest/ingest-field-studies.py:
```diff
@@ -0,0 +1,459 @@
+
+    # push current project document mapping
+
+    mapping_url = endpoint + "/dco/project/_mapping"
```
This should probably be updated to `/dco/field-study/_mapping`

nahgnaw replied:
`mapping_url` changed to `/dco/field-study/_mapping`.

zednis commented on html/field-studies.html:
```diff
@@ -0,0 +1,301 @@
+        jQuery(document).ready(function($) {
+            $('.facet-view-simple').facetview({
+                search_url: '//localhost:9200/dco/field-study/_search',
+                page_size: 20,
```
What do we think about raising the default page size for the map?  I would prefer us to attempt to show all pins on the map so I would raise this to 100 (if it does not negatively affect performance)

nahgnaw replied:
Changed to 100. I don't see any negative affect.

zednis commented on html/field-studies.html:
```diff
@@ -0,0 +1,301 @@
+    <link rel="stylesheet" href="facetview2/css/facetview.css">
+    <link rel="stylesheet" href="browsers.css">
+
+    <script id="project-template" type="text/x-handlebars-template">
```
Do we still want to show the result set like a regular browser?

nahgnaw replied:
Now I'm using css to hide the table.

nahgnaw committed:
Change mapping url to field-study in the ingest script.

nahgnaw committed:
Change result page_size to 100.

nahgnaw commented:
I looked for the popup paging functionality John talked about in leaflet, but unfortunately I didn't find any sample code. The old implementation used arcgis js library. Do we want to switch back?

zednis commented:
I am ok with switching if we can do so without losing any current functionality.

nahgnaw committed:
Switch to arcgis js library.

zednis commented:
Looking good! :+1: 

@olyerickson what do you think?

mrpatrickwest commented on html/field-studies.html:
```diff
@@ -0,0 +1,301 @@
+        jQuery(document).ready(function($) {
+            $('.facet-view-simple').facetview({
+                search_url: '//localhost:9200/dco/field-study/_search',
+                page_size: 100,
```
the arcgis version sets it to 1000. Why not just do that? Or is there a way to set it to all, like maybe 0?

nahgnaw replied:
If the arcgis version looks good, we are not gonna use this one.

nahgnaw committed:
Merge branch 'master' of github.com:tetherless-world/dco-elasticsearch into field-study-browser

nahgnaw committed:
field-study.html is now the arcgis one.

mrpatrickwest commented:
Looks good to me! :+1: 

@olyerickson just waiting for you to give the go ahead

olyerickson commented:
The Field Study Browser http://udco.tw.rpi.edu/vivo/field-studies-test
looks great!

I'm seeing (on the map) the functionality of the previous version with the
performance we expect from the new browsers based on Elasticsearch.
Awesome.

In particular,
- I see lots of field sites, making it look like a plot
- I see the paging
- I see the rich information in the pop-ups

I say, ship it!

On Mon, Oct 26, 2015 at 1:55 PM, Patrick West notifications@github.com
wrote:

> Looks good to me! [image: :+1:]
> 
> @olyerickson https://github.com/olyerickson just waiting for you to
> give the go ahead
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/tetherless-world/dco-elasticsearch/pull/25#issuecomment-151226920
> .

## 

John S. Erickson, Ph.D.
Director of Operations, The Rensselaer IDEA
Deputy Director, Web Science Research Center (RPI)
http://tw.rpi.edu olyerickson@gmail.com
Twitter & Skype: olyerickson

mrpatrickwest commented:
:shipit:

which means that @nahgnaw can merge the pull request and delete the branch, in that order of course

mrpatrickwest commented on html/field-studies.html:
```diff
@@ -0,0 +1,426 @@
+    <title>Field Studies Browser</title>
+
+    <script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/handlebars.js/4.0.1/handlebars.min.js"></script>
+    <link rel="stylesheet" href="http://js.arcgis.com/3.14/esri/css/esri.css">
```
Line 8 and line 9 need to use //js.arcgis.com instead of http://js.arcgis.com so that we don't have to worry about the protocol.

mrpatrickwest commented:
@nahgnaw no, we don't want to change this right now. We're working on making this so that it doesn't matter whether you're in production or test.

Commit:
Change links for arcgis css & js to ignore protocol

Was http://arcgis but needed to be //arcgis to ignore the protocol

Commit:
Ingest script for field studies

Be able to run the field study es ingest from the command line, pusing
the mapping, running the queries, creating the es documents, adding the
documents.
